### PR TITLE
Feat (pt): Expose Linear Ener Model 

### PR DESCRIPTION
**Pull request by anyangml**
## Summary by CodeRabbit

- **New Features**
	- Introduced two new JSON configuration files for linear energy calculations in water simulations.
	- Launched the `LinearEnergyModel` class for advanced energy and force calculations.
	- Added a parameter for customizable model weighting in the linear energy model.
	- Expanded test suite with new test classes for validating linear energy models.
	- Added new model configurations and test classes to enhance testing capabilities.

- **Bug Fixes**
	- Corrected input handling in the deserialization method for version compatibility.
	- Adjusted numerical values in data files for accurate testing.

### CODE REVIEW

**Comment by anyangml** [comment]
This PR should resolve #4186

**Review by njzjz — CHANGES_REQUESTED** [review]
Please mark they have been supported in the documentation, as well as the doc for arguments.

image
image

Please add your new file to `test_examples.py`.

Have you run it manually?

File: deepmd/pt/model/model/dp_linear_model.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+)
+
+from .dp_model import (
+    DPModelCommon,
+)
+from .make_model import (
+    make_model,
+)
+
+DPLinearModel_ = make_model(LinearEnergyAtomicModel)
```
Comment (wanghan-iapcm) [inline deepmd/pt/model/model/dp_linear_model.py:28]: `LinearEnergyModel`  should be a better name.

File: deepmd/pt/model/model/__init__.py [hunk 2]
```diff
@@ -105,6 +108,62 @@ def get_spin_model(model_params):
     return SpinEnergyModel(backbone_model=backbone_model, spin=spin)
 
 
+def get_linear_model(model_params):
+    model_params = copy.deepcopy(model_params)
+    weights = model_params.get("weights", "mean")
+    list_of_models = []
+    ntypes = len(model_params["type_map"])
+    for sub_model_params in model_params["models"]:
```
Comment (wanghan-iapcm) [inline deepmd/pt/model/model/__init__.py:116]: would be better if we can use get_standard_model to parse sub-models.

File: deepmd/pt/model/model/dp_linear_model.py (hunk 1 above)
Comment (anyangml) [reply deepmd/pt/model/model/dp_linear_model.py:28]: It is an atomic model though, are you suggesting `DPLinearModel` should be `DPLinearEnergyModel`?

**Comment by anyangml** [comment]
> Please mark they have been supported in the documentation, as well as the doc for arguments.
> 
> image image
> 
> Please add your new file to `test_examples.py`.
> 
> Have you run it manually?

Yes, I have tested 'train' 'freeze' 'finetune'.

File: deepmd/pt/model/model/__init__.py (hunk 2 above)
Comment (njzjz) [reply deepmd/pt/model/model/__init__.py:116]: Why not use `get_model` directly?

Comment (anyangml) [reply deepmd/pt/model/model/__init__.py:116]: The sub-models are at the atomic level, but as far as I understand, the `get_standard_model` method return the wrapped model directly.

Comment (njzjz) [reply deepmd/pt/model/model/__init__.py:116]: I believe that #4186 requires model-level sub-models. In detail, she wants to combine the DP model and a DMFF wrapper model, which should not be at the atomic level.

Comment (anyangml) [reply deepmd/pt/model/model/__init__.py:116]: > I believe that #4186 requires model-level sub-models. In detail, she wants to combine the DP model and a DMFF wrapper model, which should not be at the atomic level.

Do we want to keep atomic level linear models and create a new class to hold model level linear models?

**Comment by anyangml** [comment]
@njzjz We decided to not include model-level linear model in this PR.

File: examples/water/zbl/input.json [hunk 3]
```diff
@@ -10,7 +10,7 @@
       "H"
     ],
     "descriptor": {
-      "type": "se_e2_a",
+      "type": "se_atten",
```
Comment (njzjz) [inline examples/water/zbl/input.json:13]: We may use `se_atten_v2`

**Review by njzjz — APPROVED** [review]

**Review by iProzd — COMMENTED** [review]
Shall we add a linear model example using two dp atomic model?

**Review by iProzd — APPROVED** [review]

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**